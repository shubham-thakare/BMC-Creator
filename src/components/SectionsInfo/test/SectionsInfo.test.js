import React from 'react';
import { shallow } from 'enzyme';
import {
  KeyPartners,
  KeyActivities,
  KeyResources,
  ValuePropositions,
  CustomerRelationships,
  Channels,
  CustomerSegments,
  CostStructure,
  RevenueStreams,
  PopoverBody,
  PopoverQuestion,
  PopoverHintTitle,
  PopoverHintText,
} from '..';

let KeyPartnersWrapper;
let KeyActivitiesWrapper;
let KeyResourcesWrapper;
let ValuePropositionsWrapper;
let CustomerRelationshipsWrapper;
let ChannelsWrapper;
let CustomerSegmentsWrapper;
let CostStructureWrapper;
let RevenueStreamsWrapper;

describe('SectionsInfo Components', () => {
  beforeAll(() => {
    KeyPartnersWrapper = shallow(<KeyPartners />);
    KeyActivitiesWrapper = shallow(<KeyActivities />);
    KeyResourcesWrapper = shallow(<KeyResources />);
    ValuePropositionsWrapper = shallow(<ValuePropositions />);
    CustomerRelationshipsWrapper = shallow(<CustomerRelationships />);
    ChannelsWrapper = shallow(<Channels />);
    CustomerSegmentsWrapper = shallow(<CustomerSegments />);
    CostStructureWrapper = shallow(<CostStructure />);
    RevenueStreamsWrapper = shallow(<RevenueStreams />);
  });

  describe('KeyPartners Component', () => {
    it('To match snapshot of KeyPartners', async (done) => {
      expect(KeyPartnersWrapper).toMatchSnapshot();
      done();
    });
    it('KeyPartners should contain 1 PopoverBody, 4 PopoverQuestions, 1 PopoverHintTitle, and 3 PopoverHintText', async (done) => {
      expect(KeyPartnersWrapper.find(PopoverBody)).toHaveLength(1);
      expect(KeyPartnersWrapper.find(PopoverQuestion)).toHaveLength(4);
      expect(KeyPartnersWrapper.find(PopoverHintTitle)).toHaveLength(1);
      expect(KeyPartnersWrapper.find(PopoverHintText)).toHaveLength(3);
      done();
    });
  });

  describe('KeyActivities Component', () => {
    it('To match snapshot of KeyActivities', async (done) => {
      expect(KeyActivitiesWrapper).toMatchSnapshot();
      done();
    });
    it('KeyActivities should contain 1 PopoverBody, 4 PopoverQuestions, 1 PopoverHintTitle, and 3 PopoverHintText', async (done) => {
      expect(KeyActivitiesWrapper.find(PopoverBody)).toHaveLength(1);
      expect(KeyActivitiesWrapper.find(PopoverQuestion)).toHaveLength(4);
      expect(KeyActivitiesWrapper.find(PopoverHintTitle)).toHaveLength(1);
      expect(KeyActivitiesWrapper.find(PopoverHintText)).toHaveLength(3);
      done();
    });
  });

  describe('KeyResources Component', () => {
    it('To match snapshot of KeyResources', async (done) => {
      expect(KeyResourcesWrapper).toMatchSnapshot();
      done();
    });
    it('KeyResources should contain 1 PopoverBody, 4 PopoverQuestions, 1 PopoverHintTitle, and 4 PopoverHintText', async (done) => {
      expect(KeyResourcesWrapper.find(PopoverBody)).toHaveLength(1);
      expect(KeyResourcesWrapper.find(PopoverQuestion)).toHaveLength(4);
      expect(KeyResourcesWrapper.find(PopoverHintTitle)).toHaveLength(1);
      expect(KeyResourcesWrapper.find(PopoverHintText)).toHaveLength(4);
      done();
    });
  });

  describe('ValuePropositions Component', () => {
    it('To match snapshot of ValuePropositions', async (done) => {
      expect(ValuePropositionsWrapper).toMatchSnapshot();
      done();
    });
    it('ValuePropositions should contain 1 PopoverBody, 4 PopoverQuestions, 1 PopoverHintTitle, and 11 PopoverHintText', async (done) => {
      expect(ValuePropositionsWrapper.find(PopoverBody)).toHaveLength(1);
      expect(ValuePropositionsWrapper.find(PopoverQuestion)).toHaveLength(4);
      expect(ValuePropositionsWrapper.find(PopoverHintTitle)).toHaveLength(1);
      expect(ValuePropositionsWrapper.find(PopoverHintText)).toHaveLength(11);
      done();
    });
  });

  describe('CustomerRelationships Component', () => {
    it('To match snapshot of CustomerRelationships', async (done) => {
      expect(CustomerRelationshipsWrapper).toMatchSnapshot();
      done();
    });
    it('CustomerRelationships should contain 1 PopoverBody, 4 PopoverQuestions, 1 PopoverHintTitle, and 6 PopoverHintText', async (done) => {
      expect(CustomerRelationshipsWrapper.find(PopoverBody)).toHaveLength(1);
      expect(CustomerRelationshipsWrapper.find(PopoverQuestion)).toHaveLength(
        4,
      );
      expect(CustomerRelationshipsWrapper.find(PopoverHintTitle)).toHaveLength(
        1,
      );
      expect(CustomerRelationshipsWrapper.find(PopoverHintText)).toHaveLength(
        6,
      );
      done();
    });
  });

  describe('Channels Component', () => {
    it('To match snapshot of Channels', async (done) => {
      expect(ChannelsWrapper).toMatchSnapshot();
      done();
    });
    it('Channels should contain 1 PopoverBody, 6 PopoverQuestions, 6 PopoverHintTitle, and 5 PopoverHintText', async (done) => {
      expect(ChannelsWrapper.find(PopoverBody)).toHaveLength(1);
      expect(ChannelsWrapper.find(PopoverQuestion)).toHaveLength(6);
      expect(ChannelsWrapper.find(PopoverHintTitle)).toHaveLength(6);
      expect(ChannelsWrapper.find(PopoverHintText)).toHaveLength(5);
      done();
    });
  });

  describe('CustomerSegments Component', () => {
    it('To match snapshot of CustomerSegments', async (done) => {
      expect(CustomerSegmentsWrapper).toMatchSnapshot();
      done();
    });
    it('CustomerSegments should contain 1 PopoverBody, 2 PopoverQuestions, and 5 PopoverHintText', async (done) => {
      expect(CustomerSegmentsWrapper.find(PopoverBody)).toHaveLength(1);
      expect(CustomerSegmentsWrapper.find(PopoverQuestion)).toHaveLength(2);
      expect(CustomerSegmentsWrapper.find(PopoverHintText)).toHaveLength(5);
      done();
    });
  });

  describe('CostStructure Component', () => {
    it('To match snapshot of CostStructure', async (done) => {
      expect(CostStructureWrapper).toMatchSnapshot();
      done();
    });
    it('CostStructure should contain 1 PopoverBody, 3 PopoverQuestions, 2 PopoverHintTitle, and 6 PopoverHintText', async (done) => {
      expect(CostStructureWrapper.find(PopoverBody)).toHaveLength(1);
      expect(CostStructureWrapper.find(PopoverQuestion)).toHaveLength(3);
      expect(CostStructureWrapper.find(PopoverHintTitle)).toHaveLength(2);
      expect(CostStructureWrapper.find(PopoverHintText)).toHaveLength(6);
      done();
    });
  });

  describe('RevenueStreams Component', () => {
    it('To match snapshot of RevenueStreams', async (done) => {
      expect(RevenueStreamsWrapper).toMatchSnapshot();
      done();
    });
    it('RevenueStreams should contain 1 PopoverBody, 5 PopoverQuestions, and 1 Table', async (done) => {
      expect(RevenueStreamsWrapper.find(PopoverBody)).toHaveLength(1);
      expect(RevenueStreamsWrapper.find(PopoverQuestion)).toHaveLength(5);
      expect(RevenueStreamsWrapper.find('table')).toHaveLength(1);
      done();
    });
  });
});
