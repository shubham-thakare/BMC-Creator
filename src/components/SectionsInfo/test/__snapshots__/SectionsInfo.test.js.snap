// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionsInfo Components Channels Component To match snapshot of Channels 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    Through which channels do our customer segments want to be reached?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How are we reaching them now?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How are our channels integrated?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which ones work best?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which ones are most cost-efficient?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How are we integrating them with customer routines?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintTitle>
    CHANNEL PHASES
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintTitle>
    1. Awareness
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    How do we raise awareness about our company's products and services?
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintTitle>
    2. Evaluation
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    How do we help customers evaluate our organization's value proposition?
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintTitle>
    3. Purchase
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    How do we allow customers to purchase specific products and services?
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintTitle>
    4. Delivery
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    How do we deliver a value proposition to customers?
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintTitle>
    5. After sales
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    How do we provide post-purchase customer support?
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components CostStructure Component To match snapshot of CostStructure 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    What are the most important costs inherent in our business model?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which key resources are most expensive?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which key activities are most expensive?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintTitle>
    IS YOUR BUSINESS MORE
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    Cost driven (leanest cost structure, low price value proposition, maximum automation, extensive outsourcing)
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Value driven (focused on value creation, premium value proposition)
  </Commonstyle__PopoverHintText>
  <br />
  <Commonstyle__PopoverHintTitle>
    SAMPLE CHARACTERISTICS
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    Fixed costs (salaries, rents, utilities)
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Variable costs
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Economies of scale
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Economies of scope
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components CustomerRelationships Component To match snapshot of CustomerRelationships 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    What type of relationship does each of our customer segments expect us to establish and maintain with them?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which ones have we established?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How are they integrated with the rest of our business model?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How costly are they?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintTitle>
    EXAMPLES
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    Personal assistance
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Dedicated personal assistance
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Self-service
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Automated services
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Communities
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Co-creation
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components CustomerSegments Component To match snapshot of CustomerSegments 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    For whom are we creating value?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Who are our most important customers?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintText>
    Mass market
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Niche market
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Segmented
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Diversified
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Multi-sided platform
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components KeyActivities Component To match snapshot of KeyActivities 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    What key activities do our value propositions require?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Our distribution channels?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Customer relationships?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Revenue streams?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintTitle>
    Categories
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    Production
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Problem solving
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Platform/Network
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components KeyPartners Component To match snapshot of KeyPartners 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    Who are our key partners?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Who are our key suppliers?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which key resources are we acquairing from partners?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which key activities do partners perform?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintTitle>
    Hints for partnerships
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    Optimization and economy
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Reduction of risk and uncertainity
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Acquisition of particular resources and activities
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components KeyResources Component To match snapshot of KeyResources 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    What key resources do our value propositions require?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Our distribution channels?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Customer relationships?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Revenue streams?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintTitle>
    Types of resources
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    Physical
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Intellectual (brand patents, copyrights, data)
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Human
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Financial
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components RevenueStreams Component To match snapshot of RevenueStreams 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    For what value are our customers really willing to pay?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    For what do they currently pay?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How are they currently paying?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How would they prefer to pay?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    How much does each revenue stream contribute to overall revenues?
  </Commonstyle__PopoverQuestion>
  <br />
  <table>
    <thead>
      <th>
        Types
      </th>
      <th>
        Fixed Pricing
      </th>
      <th>
        Dynamic Pricing
      </th>
    </thead>
    <tbody>
      <tr>
        <td>
          Asset sale
        </td>
        <td>
          List price
        </td>
        <td>
          Negotiation (bargaining)
        </td>
      </tr>
      <tr>
        <td>
          Usage fee
        </td>
        <td>
          Product feature dependent
        </td>
        <td>
          Yield management
        </td>
      </tr>
      <tr>
        <td>
          Subscription fees
        </td>
        <td>
          Customer segment
        </td>
        <td>
          Real-time-market
        </td>
      </tr>
      <tr>
        <td>
          Lending/Renting/Leasing
        </td>
        <td>
          dependent
        </td>
      </tr>
      <tr>
        <td>
          Licensing
        </td>
        <td>
          Volume dependent
        </td>
      </tr>
      <tr>
        <td>
          Brokerage fees
        </td>
      </tr>
      <tr>
        <td>
          Advertising
        </td>
      </tr>
    </tbody>
  </table>
</Commonstyle__PopoverBody>
`;

exports[`SectionsInfo Components ValuePropositions Component To match snapshot of ValuePropositions 1`] = `
<Commonstyle__PopoverBody>
  <Commonstyle__PopoverQuestion>
    What value do we deliver to the customer?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which one of our customer's problems are we helping to solve?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    What bundles of products and services are we offering to each customer segment?
  </Commonstyle__PopoverQuestion>
  <Commonstyle__PopoverQuestion>
    Which customer needs are we satisfying?
  </Commonstyle__PopoverQuestion>
  <br />
  <Commonstyle__PopoverHintTitle>
    CHARACTERISTICS
  </Commonstyle__PopoverHintTitle>
  <Commonstyle__PopoverHintText>
    Newness
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Performance
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Customization
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Getting the job done
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Design
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Brand/Status
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Price
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Cost reduction
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Risk reduction
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Accessibility
  </Commonstyle__PopoverHintText>
  <Commonstyle__PopoverHintText>
    Convenience/Usability
  </Commonstyle__PopoverHintText>
</Commonstyle__PopoverBody>
`;
